Necrosis/XML/Attributes.lua

```diff
@@ -226,11 +234,15 @@ function Necrosis:PetSpellAttribute()
 	for i = 2, 6, 1 do
 		local f = _G["NecrosisPetMenu"..i]
 		if f then
+			local SpellName_Rank = self.Spell[i+1].Name
+			if self.Spell[i+1].Rank and not (self.Spell[i+1].Rank == " ") then
+				SpellName_Rank = SpellName_Rank.."("..self.Spell[i+1].Rank..")"
+			end
```

Comment:
These 4 lines appear multiple times. let's abstract this into a helper method.

Reply:
```
function Necrosis:SpellNameWithRank(spellId)
    local SpellName_Rank = self.Spell[spellId].Name
    if self.Spell[i+1].Rank and not (self.Spell[spellId].Rank == " ") then
        SpellName_Rank = SpellName_Rank.."("..self.Spell[spellId].Rank..")"
    end
    SpellName_Rank 
end

local SpellName_Rank = SpellNameWithRank(i+1)

Reply:
it looks like this new method could be used in other place, I will comment on them in this PR.